packages/sdk-core/src/modules/metrics/MetricsSubmissionQueue.ts

```diff
@@ -44,13 +45,16 @@ export class MetricsSubmissionQueue<T extends MetricsEvent> implements MetricsQu
         for (let attempts = 0; attempts < this.MAXIMUM_NUMBER_OF_ATTEMPTS; attempts++) {
             const response = await this._requestHandler.post(
                 this._submissionUrl,
-                JSON.stringify({
-                    ...this._metricMetadata,
-                    [this._eventName]: events,
-                    metadata: {
-                        dropped_events: this._numberOfDroppedRequests,
+                JSON.stringify(
+                    {
+                        ...this._metricMetadata,
+                        [this._eventName]: events,
+                        metadata: {
+                            dropped_events: this._numberOfDroppedRequests,
+                        },
                     },
-                }),
+                    jsonEscaper(),
+                ),
```

Comment:
Maybe stringify once before the `for` loop?

Reply:
good idea. Thx!